source/fe/fe_poly_tensor.cc

```diff
@@ -426,162 +441,68 @@ FE_PolyTensor<POLY,dim,spacedim>::fill_fe_values (
   else if (mapping_type == mapping_nedelec)
     get_face_sign_change_nedelec (cell, this->dofs_per_face, sign_change);
 
+  // we only need to update element data if the current cell is not a
+  // translation of the previous one; or, even if it is a translation, if we
```

Comment:
There are a lot of changes in this function, to the point where it's difficult to read the patch. Do I see this right that you're just switching the inner loops and the outer switch statement?